lang/c/src/datum.c

```diff
@@ -156,7 +156,7 @@ static avro_datum_t avro_bytes_private(char *bytes, int64_t size,
 avro_datum_t avro_bytes(const char *bytes, int64_t size)
 {
 	char *bytes_copy = (char *) avro_malloc(size);
-	if (!bytes_copy) {
+	if (!bytes_copy && size) {
```

Comment:
What would happen now if `size` is `0`?
It looks to me that no matter what is the value of `bytes_copy` it will never enter the `if` and it will call `memcpy(bytes_copy, bytes, 0)`